### PR TITLE
MEPTS 1160 | TX_TB Changes For March 2023

### PR DESCRIPTION
**Pull request by TinoMuianga**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxTbMonthlyCascadeReport.java [hunk 1]
```diff
@@ -32,12 +32,12 @@ public String getUuid() {
 
   @Override
   public String getName() {
-    return "TX TB Monthly Cascade Report";
+    return "PEPFAR MER 2.6.1 Semi-Annual";
   }
 
   @Override
   public String getDescription() {
-    return "TX TB Monthly Cascade Report to PEPFAR";
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxTbMonthlyCascadeReport.java]: Update the TxTb Setup

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java [hunk 2]
```diff
@@ -692,6 +692,49 @@ public CohortDefinition positiveInvestigationResultComposition() {
...
+   *
+   * @return {@link CohortDefinition}
+   */
+  public CohortDefinition negativeInvestigationResultComposition() {
+    CompositionCohortDefinition cd = new CompositionCohortDefinition();
+    CohortDefinition N = negativeInvestigationResult();
+    cd.addSearch("N", map(N, codedObsParameterMapping));
+    cd.setCompositionString("N");
+    addGeneralParameters(cd);
+    return cd;
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java]: No need to create composition without preposition.

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java [hunk 3]
```diff
@@ -1308,112 +1371,134 @@ public CohortDefinition getPatientsTransferredOut() {
...
+            + "                                                       AND        o.voided = 0 "
+            + "                                                       AND        e.encounter_type = ${masterCardDrugPickupEncounterType} "
+            + "                                                       AND        o.concept_id = ${artDatePickup} "
+            + "                                                       AND        o.value_datetime BETWEEN :startDate AND        :endDate "
+            + "                                                       AND        e.location_id = :location "
+            + "                                                       GROUP BY   p.patient_id ) pickup_arv "
+            + "                                 ON         pickup_arv.patient_id = transferred_out.patient_id "
+            + "                                 WHERE      transferred_out.last_date > date_add(latest_fila.value_datetime, interval 1 day) "
+            + "                                 AND        transferred_out.last_date <= date_add(pickup_arv.value_datetime, interval 31 day) "
+            + "             GROUP BY transferred_out.patient_id";
```
Comment (eliazmuchave) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TXTBCohortQueries.java]: This query is not in accordance with the specification

**Review by eliazmuchave — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxTbMonthlyCascadeReport.java (hunk 1 above)
Comment (TinoMuianga) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/reports/SetupTxTbMonthlyCascadeReport.java]: Solved

**Review by eliazmuchave — APPROVED** [review]

**Pull request merged**